include/revng/StackAnalysis/SegregateDirectStackAccesses.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/BasicAnalyses/GeneratedCodeBasicInfo.h"
+#include "revng/Support/revng.h"
+
+class SegregateDirectStackAccessesPass
+  : public llvm::PassInfoMixin<SegregateDirectStackAccessesPass> {
+  llvm::LLVMContext *Context = nullptr;
+  GeneratedCodeBasicInfo *GCBI = nullptr;
+  std::vector<llvm::Instruction *> DirectStackAccesses;
+  std::vector<llvm::Instruction *> NotDirectStackAccesses;
```

Comment:
Passes should not have any fields unless they are exposing results of an analysis.
These fields represent temporary data whose lifetime should not exceed the `run` method.

Move all fields and private methods into an `Impl` class in the `.cpp` file.

The run method will then look like:
```c++
llvm::PreservedAnalyses
SegregateDirectStackAccessesPass::run(llvm::Function &F, llvm::FunctionAnalysisManager &FAM) {
  SegregateDirectStackAccessesPassImpl SDAP;
  SDAP.run();
}
```

Reply:
Sounds good.

Reply:
This requires to pass `FAM` to the constructor of `SegregateDirectStackAccessesPassImpl` though, which does not seem to be that clean.

Reply:
It seems fine to me.